### ISSUE TITLE
Type 'ListenEvents' does not satisfy the constraint 'EventsMap'. `node_modules/socket.io/dist/namespace.d.ts`

### ISSUE THREAD

**Issue by marc-wittwer** [socketio/socket.io#4459]
**Describe the bug**
I get the following error when I run `npm run build` in my node.js express application.

```
 npm run build

> express-api-2@0.9.0 build
> tsc -p .

node_modules/socket.io/dist/namespace.d.ts:14:133 - error TS2344: Type 'ListenEvents' does not satisfy the constraint 'EventsMap'.

14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                                                                                                       ~~~~~~~~~~~~

  node_modules/socket.io/dist/namespace.d.ts:14:42
    14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                ~~~~~~~~~~~~
    This type parameter might need an `extends EventsMap` constraint.

node_modules/socket.io/dist/namespace.d.ts:15:42 - error TS2344: Type 'ListenEvents' does not satisfy the constraint 'EventsMap'.

15     new_namespace: (namespace: Namespace<ListenEvents, EmitEvents, ServerSideEvents, SocketData>) => void;
                                            ~~~~~~~~~~~~

  node_modules/socket.io/dist/namespace.d.ts:14:42
    14 export interface ServerReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> extends NamespaceReservedEventsMap<ListenEvents, EmitEvents, ServerSideEvents, SocketData> {
                                                ~~~~~~~~~~~~
    This type parameter might need an `extends EventsMap` constraint.


Found 2 errors in the same file, starting at: node_modules/socket.io/dist/namespace.d.ts:14
```

My system:
Node v17.9.1 (npm v8.11.0)
Ubuntu 20.04
`socket.io: "^4.5.1",`

**Comment by marc-wittwer** [comment]
https://github.com/socketio/socket.io/pull/4431
This might fix it.

**Comment by shhdharmen** [comment]
For time being, I used commonJS pattern of `require` and it's building fine:

```typescript
const { Server } = require("socket.io");
```

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io/pull/4431, included in version [4.5.2](https://github.com/socketio/socket.io/releases/tag/4.5.2).

Thanks for reporting this issue :+1:
